lib/reek/rake/task.rb

```diff
@@ -32,118 +33,83 @@ module Rake
     #   rake reek REEK_OPTS=-s                   # sorts the report by smell
     #
     class Task < ::Rake::TaskLib
-      # Name of reek task.
-      # Defaults to :reek.
-      attr_accessor :name
+      # Name of reek task. Defaults to :reek.
+      attr_writer :name
 
-      # Array of directories to be added to $LOAD_PATH before running reek.
-      # Defaults to ['<the absolute path to reek's lib directory>']
-      attr_accessor :libs
-
-      # Glob pattern to match config files.
+      # Path to reeks config file.
       # Setting the REEK_CFG environment variable overrides this.
-      # Defaults to 'config/**/*.reek'.
-      attr_accessor :config_file
+      attr_writer :config_file
 
+      # Here for backwards compatibility - can be removed in reek 2.
       def config_files=(value)
         @config_file = value
       end
 
-      def config_files
-        @config_file
-      end
-
       # Glob pattern to match source files.
       # Setting the REEK_SRC environment variable overrides this.
       # Defaults to 'lib/**/*.rb'.
-      attr_accessor :source_files
+      attr_writer :source_files
 
       # String containing commandline options to be passed to Reek.
       # Setting the REEK_OPTS environment variable overrides this value.
       # Defaults to ''.
-      attr_accessor :reek_opts
-
-      # Array of commandline options to pass to ruby. Defaults to [].
-      attr_accessor :ruby_opts
+      attr_writer :reek_opts
 
       # Whether or not to fail Rake when an error occurs (typically when smells are found).
       # Defaults to true.
-      attr_accessor :fail_on_error
+      attr_writer :fail_on_error
 
       # Use verbose output. If this is set to true, the task will print
       # the reek command to stdout. Defaults to false.
-      attr_accessor :verbose
+      attr_writer :verbose
 
-      # Defines a new task, using the name +name+.
       def initialize(name = :reek)
-        @name = name
-        @libs = [File.expand_path(File.dirname(__FILE__) + '/../../../lib')]
-        @config_file = nil
-        @source_files = nil
-        @ruby_opts = []
-        @reek_opts = ''
+        @name          = name
+        @reek_opts     = ''
         @fail_on_error = true
-        @sort = nil
+        @source_files  = 'lib/**/*.rb'
+        @verbose       = false
 
         yield self if block_given?
-        @config_file ||= Pathname.glob('config/**/*.reek').detect(&:file?)
```

Comment:
You will have to somehow restore this globbing as well if you want to merge this for 1.6.2.